src/libNode/Node.cpp

```diff
@@ -997,6 +997,43 @@ bool Node::ProcessTxnPacketFromLookupCore(const vector<unsigned char>& message,
   return true;
 }
 
+bool Node::ProcessProposeGasPrice(
+    [[gnu::unused]] const vector<unsigned char>& message,
+    [[gnu::unused]] unsigned int offset, [[gnu::unused]] const Peer& from) {
+  LOG_MARKER();
+
+  if (LOOKUP_NODE_MODE) {
+    LOG_GENERAL(WARNING,
+                "Node::ProcessProposeGasPrice not expected to "
+                "be called from LookUp node.");
+    return true;
+  }
+
+  if (IsMessageSizeInappropriate(message.size(), offset, UINT256_SIZE)) {
+    LOG_GENERAL(WARNING,
+                "Message size for ProcessProposeGasPrice is too short");
+    return false;
+  }
+
+  if (string(from.GetPrintableIPAddress()) != "127.0.0.1") {
```

Comment:
what is the rationale for this check?

Reply:
The current check is only allow the user send the command from local machine. Yup I agree we can configure that in the constants so let the user to decide which ip is legal to propose gas price for it.

Reply:
After discussing with @kaikawaliu, another option is we can have the node's user affix a signature to the end of this message, so in effect the node verifies its own signature.